test/test_invalid_headers.py

```diff
@@ -239,20 +241,33 @@ class TestFilter(object):
     ]
 
     hdr_validation_combos = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, True),
-            (True, False),
-            (False, True),
-            (False, False)
-        ]
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in (
+            itertools.product([True, False], repeat=3)
+        )
     ]
 
     hdr_validation_no_trailers = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, False),
-            (False, False)
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in [
+            (True, False, False),
+            (False, False, False),
+        ]
+    ]
+
+    hdr_validation_response_headers = [
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in [
+            (True, False, True),
+            (False, True, True),
+            (True, True, True),
+            (False, False, True)
```

Comment:
Is it worth generating these two lists by using `filter`? Both `no_trailers` and `response_headers` are sub-cases of the `hdr_validation_combos`, so maybe we should stop hand-coding them and instead define their condition as a filter predicate.

That is: `hdr_validation_response_headers = list(filter(lambda flags: flags.is_response_header, hdr_validation_combos))`, and `hdr_validation_no_trailers = list(filter(lambda flags: not flags.is_trailer, hdr_validation_combos))`. That way, adding more flags to the header validation flags won't force us to keep updating this test.